harvester_e2e_tests/apis/test_support_bundle.py

```diff
@@ -0,0 +1,109 @@
+import pytest
+from harvester_e2e_tests import utils
+
+pytest_plugins = [
+    'harvester_e2e_tests.fixtures.api_endpoints',
+    'harvester_e2e_tests.fixtures.user',
+    'harvester_e2e_tests.fixtures.session',
+    'harvester_e2e_tests.fixtures.support_bundle'
+]
+
+
+def test_create_volume_missing_description(
+        admin_session, harvester_api_endpoints):
+    request_json = utils.get_json_object_from_template('support_bundle')
+    del request_json['spec']['description']
+    resp = admin_session.post(
+        harvester_api_endpoints.create_support_bundle, json=request_json)
+    assert resp.status_code == 422
+    assert 'is invalid: spec.description: Required value' in resp.text
+
+
+@pytest.mark.p1
+@pytest.mark.slow
+def test_creates_successful_bundle_and_downloads(
+        admin_session, harvester_api_endpoints):
+    """
+    SupportBundle generic
+    Covers:
+        The support bundle creating
+        The support bundle generation and progress, progressing
+        The support bundle being able to be downloaded
+        The file structure existing in the support bundle correctly in logs
+    """
+    request_json = utils.get_json_object_from_template('support_bundle')
+    resp = admin_session.post(
+        harvester_api_endpoints.create_support_bundle, json=request_json)
+    assert resp.status_code == 201
+    resp_json = json.loads(resp.text)
+    name_to_use_for_download = resp_json.get('metadata').get('name')
+    progress = 0
+
+    def _wait_for_progress_to_be_ready():
+        nonlocal progress
+        resp_progress_check = admin_session.get(
+            harvester_api_endpoints.view_support_bundle +
+            name_to_use_for_download)
+        json_progress_result = json.loads(resp_progress_check.text)
+        progress = json_progress_result.get('status', {}).get('progress', 0)
+        if progress != 100:
+            return False
+        return True
+
+    successful_support_bundle_generation = polling2.poll(
+        _wait_for_progress_to_be_ready,
+        step=5,
+        timeout=300
+    )
+    assert successful_support_bundle_generation, ('Timed out waiting for ' +
+                                                  'support bundle geneartion')
+    tempdir = tempfile.mkdtemp()
+    zip_file_location = os.path.join(tempdir, 'support_bundle.zip')
+    resp_download = admin_session.get(
+        harvester_api_endpoints.download_support_bundle +
+        name_to_use_for_download +
+        '/download')
+    with open(zip_file_location, 'wb') as support_zip_file:
+        support_zip_file.write(resp_download.content)
+    list_of_files_in_support_bundle = None
+    with zipfile.ZipFile(zip_file_location, 'r') as support_zip_file:
+        list_of_files_in_support_bundle = support_zip_file.namelist()
+    cattle_fleet_local_system_fleet_agent_log = False
+    cattle_fleet_system_fleet_controller_log = False
+    cattle_fleet_system_gitjob_log = False
+    for file in list_of_files_in_support_bundle:
```

Comment:
In here, I would suggest to use regex.match:
```python
patterns = [r"logs/cattle-fleet-local-system/fleet-agent-.*/fleet-agent.log",
            r"logs/cattle-fleet-system/fleet-controller-.*/fleet-controller.log",
            r"logs/cattle-fleet-system/gitjob-.*/gitjob.log"]
# or the simple one: patterns: [r".*cattle-fleet-system.*fleet-agent.log", ...]
# I would recommend that one file use one pattern, it would be more easy to update.
matches = []
for f in files:
    for pattern in patterns:
        matches.extend([f] if re.match(pattern, f) else [])

errmsg  = ("Some file(s) not found,"
           f"files: {matches}\n"
           f"patterns: {patterns}")
assert len(matches) == len(patterns), errmsg
```

And we can just add new pattern if there have any new file need to be verified.

Reply:
I'll investigate the implementation of the regex :smile: 

I do like how that it's fairly easy to read through what is taking place without the regex implementation.  

We do end up with two nested for loops if we loop through every file and every pattern.  But I don't see that being a problem :smile: 

You're right that the extensibility with the `patterns` - should support bundle in the future implement additional things, we'd want to be able to extend this test.

Reply:
the extensibility is one of the reason to use regex, another reason is, consider we are verifying multiple files, and there is no sub-test functionality in pytest, so in this case, we are unable to catch multiple files missed.

Reply:
:smile:  - definitely :+1: 

I did realize we needed to adjust the regex a little bit.
After some testing, for our 'gitjob' regex for instance we would need to adjust it from:
```
r"logs/cattle-fleet-system/gitjob-.*/gitjob.log"
```
to:
```
r"^.*/logs/cattle-fleet-system/gitjob-.*/gitjob.log"
```
in order to match/capture the test string of:
```
supportbundle_87739721-c996-40f9-9317-3ab213cb3fc1_2022-06-22T18-29-15Z/logs/cattle-fleet-system/gitjob-cc9948fd7-6t7q2/gitjob.log
```
since `zip_file.namelist()` prints out the full path, top level directory included.